docs/user-guide/rug/tests.md

```diff
@@ -142,44 +203,140 @@ features and scenarios.
-functions to find editors and generators, query the number of
-modifications made, check if the parameters passed in were valid, see
-if the executing Rug succeeded, etc.
-
-The scenario world is an optional second parameter in the function
-that implements all steps.  For example, the declaration of the
-`#!typescript Given` function looks like this:
+editors, the world provided to the function implementing the scenario
+steps is a `#!typescript ProjectScenarioWorld`.  The `#!typescript
+ProjectScenarioWorld` interface extends `#!typescript ScenarioWorld`,
+adding functions to find and execute editors and generators, query the
+number of modifications made, and see if editing succeeded.  The
+scenario world is an optional second parameter to the function that
+implements each scenario step.  For example, the declaration of the
+`#!typescript Given` function is effectively:
 
 ```typescript
-export function Given(s: string, f: (Project, ScenarioWorld?) => void);
+import { ProjectScenarioWorld } from "@atomist/rug/test/project/Core";
+export function Given(s: string, f: (Project, ProjectScenarioWorld?) => void);
 ```
 
+In practice, what you do with the scenario's world depends on what
+*phase* of the scenario you are in.  Each BDD testing scenario has
+three phases, mapping directly to the different types of Gherkin
+steps:
+
+1.  [Setup](#setup): the `#!gherkin Given` steps are used to set up
+    testing prerequisites within the world
+2.  [Execution](#execution): the action being tested is executed on
+    the world in the `#!gherkin When` step
+3.  [Assertions](#assertions): the state of the scenario's world is
+    interrogated in the `#!gherkin Then` steps
+
+In the following sections, we will delve into each of these phases and
+describe in detail how the scenario's world can be used in each phase
+to enable testing.
+
 ## Setup
 
-The `#!gherkin Given` step is used to set up a project and world for
-testing.  Often the setup involves either using
-a ["well-known step"](#well-known-steps) or a series of steps each
-creating a file to populate the project for testing.  As an example,
-consider the `#!gherkin Given` steps below.
+The `#!gherkin Given` step is used to set up a world for a testing
+scenario.  Many testing scenarios can be entirely setup using
+a ["well-known step"][well-known], but sometimes you need to customize
+for your particular scenario.
+
+[well-known]: #well-known-steps (Well-Known BDD Steps)
+
+### Handler setup
+
+When testing command handlers, you almost always use the "nothing"
+well-known step that, as you might expect, does nothing.
+
+```gherkin
+Given nothing
+```
+
+As with all well-known steps, you do not need to supply an
+implementation, one is already available to the Rug runtime.
+
+To avoid unwanted side effects during testing, the testing framework
+will only consider executing event handlers that are explicitly
+registered within a scenario.  Thus, the setup for event handler
```

Comment:
I think "in" is closer than "with", but I still think "within" is the most appropriate.